server/dive_server/serializers/viame.py

```diff
@@ -159,13 +159,67 @@ def _parse_row_for_tracks(row: List[str]) -> Tuple[Feature, Dict, Dict, List]:
     return feature, attributes, track_attributes, confidence_pairs
 
 
-def load_csv_as_tracks(rows: List[str]) -> Dict[str, dict]:
+def create_attributes(
+    metadata_attributes: Dict[str, Attribute],
+    test_vals: Dict[str, int],
+    atr_type: str,
+    key: str,
+    val,
+):
+    valstring = f'{val}'
+    if f'{atr_type}_{key}' not in metadata_attributes:
+        metadata_attributes[f'{atr_type}_{key}'] = {
+            'belongs': atr_type,
+            'datatype': 'text',
+            'name': key,
+            'key': f'{atr_type}_{key}',
+        }
+        test_vals[f'{atr_type}_{key}'] = {}
+        test_vals[f'{atr_type}_{key}'][valstring] = 1
+    elif (
+        f'{atr_type}_{key}' in metadata_attributes and f'{atr_type}_{key}' in test_vals
+    ):
+        if valstring in test_vals[f'{atr_type}_{key}']:
+            test_vals[f'{atr_type}_{key}'][valstring] += 1
+        else:
+            test_vals[f'{atr_type}_{key}'][valstring] = 1
+
+
+def calculate_attribute_types(
+    metadata_attributes: Dict[str, Attribute], test_vals: Dict[str, int]
+):
+    for attributeKey in metadata_attributes.keys():
+        if attributeKey in test_vals:
+            attribute_type = 'number'
+            low_count = 3
+            values = []
+            for (key, val) in test_vals[attributeKey].items():
+                if val <= low_count:
+                    low_count = val
+                values.append(key)
+                if attribute_type == 'number':
+                    try:
+                        float(key)
+                    except ValueError:
+                        attribute_type = 'boolean'
+                if attribute_type == 'boolean' and key != 'True' and key != 'False':
+                    attribute_type = 'text'
+            # If all text values are used 3 or more times they are defined values
```

Comment:
If you have 100 values and 95 of them are the same and the 5 others are unique it then becomes a free-form text value.  The code there ensures that all values are used at least 3 times before it swaps itself to predefined.  So all unique values must be used at least 3 times. 
`TestAttribute` of type `text`
- value 1: 10 times
- value 2: 3 times
- value 3: 24 times
- value 4: 2 times

Because 'value 4' is only used 2 times, `TestAttribute` will be free form text instead of predefined values.

 This still isn't the best way to handle this because I'm trying to guess at intention from the values provided.  

I never questioned why I even imported this over from the previous attribute section.  The viame-csv format has no support for predefined values and its a 'feature' that was only in the UI.  So maybe we should just remove predefined values or they have to be manually set by the user.  May be a question for Matt.